mutex.go

```diff
@@ -78,8 +89,15 @@ type mutex struct {
 	client  *http.Client
 }
 
-// Lock waits indefinitely to acquire a global mutex lock.
+// Lock waits indefinitely to acquire a global mutex.
```

Comment:
I would even remove "global". Makes an assumption not relevant to the implementation.